roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/event/CarCouldLeaveSectionEvent.java

```diff
@@ -1,18 +1,66 @@
 package at.fhv.itm3.s2.roundabout.event;
 
-import at.fhv.itm3.s2.roundabout.api.event.ICarCouldLeaveSectionEvent;
+import at.fhv.itm3.s2.roundabout.RoundaboutModel;
+import at.fhv.itm3.s2.roundabout.api.entity.IStreetSection;
+import at.fhv.itm3.s2.roundabout.entity.StreetSection;
 import co.paralleluniverse.fibers.SuspendExecution;
-import desmoj.core.simulator.Entity;
 import desmoj.core.simulator.Event;
 import desmoj.core.simulator.Model;
+import desmoj.core.simulator.TimeSpan;
 
-public class CarCouldLeaveSectionEvent extends Event<Entity> implements ICarCouldLeaveSectionEvent {
+import java.util.concurrent.TimeUnit;
 
-    public CarCouldLeaveSectionEvent(Model model, String s, boolean b) {
-        super(model, s, b);
+public class CarCouldLeaveSectionEvent extends Event<StreetSection> {
+
+    /**
+     * A reference to the RoundaboutModel the CarCouldLeaveSectionEvent is part of
+     */
+    private RoundaboutModel myModel;
+
+    /**
+     * Constructs a new CarCouldLeaveSectionEvent
+     *
+     * @param model         the model this event belongs to
+     * @param name          this event's name
+     * @param showInTrace   flag to indicate if this event shall produce output for the trace
+     */
+    public CarCouldLeaveSectionEvent(Model model, String name, boolean showInTrace) {
+        super(model, name, showInTrace);
+        myModel = (RoundaboutModel)model;
```

Comment:
is RoundaboutModel needed here?
if yes, instanceof check before cast, otherwise IllegalArgumentException

Reply:
will handle it.